SIDFactoryII/source/runtime/editor/datasources/datasource_directory.cpp

```diff
@@ -158,8 +159,27 @@ namespace Editor
                 std::error_code error_code;
                 if (is_directory(path, error_code))
                     m_List.push_back({ DirectoryEntry::Folder, path });
-				else if (is_regular_file(path, error_code))
-					files.push_back(path);
+				else if (is_regular_file(path, error_code)) {
+					std::string extension = fs::path(path).extension();
+
+					// filter out files with extensions on the hide list
+					bool showFile = true;
+					for (size_t i = 0; i < m_ExtensionFilter.size(); ++i) {
+						if (extension.compare(m_ExtensionFilter[i]) == 0) {
+							showFile = false;
+						}
+					}
+
```

Comment:
You could use std::find_if instead, but this works too.

Let's fix the scopes so that it's not java-style... So the open brackets go on the next line.

Reply:
I ran clang-format, it cleaned up some more things